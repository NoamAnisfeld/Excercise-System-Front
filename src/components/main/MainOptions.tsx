import { Grid } from "@mui/material"
import FadeIn from "../FadeIn"
import MainOptionsButton from "./MainOptionsButton";
import { useViewerIsStaff } from "../../hooks";

export default function MainOptions() {

    const viewerIsStaff = useViewerIsStaff();

    return (
        <FadeIn>
            <Grid container justifyContent="center" p={1}>
                <Grid item>
                    <MainOptionsButton title="כל הקורסים" linkTo="courses" />
                </Grid>
                <Grid item>
                    <MainOptionsButton title="כל התרגילים" linkTo="assignments" />
                </Grid>
                <Grid item>
                    <MainOptionsButton title="כל ההגשות" linkTo="submissions" />
                </Grid>
                {viewerIsStaff ?
                    <Grid item>
                        <MainOptionsButton title="כל המשתמשים" linkTo="users" />
                    </Grid>
                    :
                    undefined
                }
            </Grid>
        </FadeIn>
    );
}