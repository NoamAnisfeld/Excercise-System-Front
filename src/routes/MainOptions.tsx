import { Grid } from "@mui/material"
import MyCoursesButton from "../components/MyCoursesButton"
import FadeIn from "../components/FadeIn"

export default function MainOptions() {
    return (
        <FadeIn>
            <Grid container justifyContent="center">
                <Grid item>
                    <MyCoursesButton />
                </Grid>
            </Grid>
        </FadeIn>
    );
}