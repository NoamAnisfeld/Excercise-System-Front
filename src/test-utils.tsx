import setupStore from './global-state/store.ts'
import { Provider as ReduxProvider } from 'react-redux'
import { createBrowserRouter, RouterProvider } from "react-router-dom"
import { ReactNode } from "react"

export function wrapWithStore(node: ReactNode) {
    const store = setupStore();

    return (
        <ReduxProvider store={store}>
            {node}
        </ReduxProvider>
    );
}

export function wrapWithRouter(element: ReactNode) {
    const router = createBrowserRouter([
        {
            path: '/*',
            element,
        },
    ]);
    
    return <RouterProvider router={router} />;
}