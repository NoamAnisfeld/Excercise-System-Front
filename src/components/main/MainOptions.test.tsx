import { test, expect } from 'vitest'
import { wrapWithStore, wrapWithRouter } from '../../test-utils'
import { render } from '@testing-library/react'
import MainOptions from './MainOptions'

test('MainOptions', () => {
    const result = render(wrapWithStore(wrapWithRouter(<MainOptions />)))
    expect(result).toMatchSnapshot();
})