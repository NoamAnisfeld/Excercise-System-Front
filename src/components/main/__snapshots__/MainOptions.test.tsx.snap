// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MainOptions 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-0"
        style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
      >
        <div
          class="MuiGrid-root MuiGrid-container css-u98wuq-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
                href="/courses"
                tabindex="0"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                    >
                      כל הקורסים
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
                href="/assignments"
                tabindex="0"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                    >
                      כל התרגילים
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <div
              class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
            >
              <a
                class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
                href="/submissions"
                tabindex="0"
              >
                <div
                  class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
                >
                  <div
                    class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                  >
                    <span
                      class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                    >
                      כל ההגשות
                    </span>
                  </div>
                </div>
                <span
                  class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
                />
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-0"
      style="opacity: 1; webkit-transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms; transition: opacity 225ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-u98wuq-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
              href="/courses"
              tabindex="0"
            >
              <div
                class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                  >
                    כל הקורסים
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
              href="/assignments"
              tabindex="0"
            >
              <div
                class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                  >
                    כל התרגילים
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1ftfsnj-MuiPaper-root-MuiCard-root"
          >
            <a
              class="MuiButtonBase-root MuiCardActionArea-root css-1oitmrd-MuiButtonBase-root-MuiCardActionArea-root"
              href="/submissions"
              tabindex="0"
            >
              <div
                class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
              >
                <div
                  class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
                >
                  <span
                    class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1koc0za-MuiTypography-root"
                  >
                    כל ההגשות
                  </span>
                </div>
              </div>
              <span
                class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
              />
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </a>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
